src/models/documents.test.ts

```diff
@@ -46,33 +41,107 @@ describe("documents model", () => {
     expect(documents.getSectionDocument("1", "2")).toBe(document);
   });
 
-  const getPublishedDocument = (createdAt: number, sectionId: string, groupId: string) => {
-    return DocumentModel.create({
-      uid: "1",
-      type: "publication",
-      key: `llDoc-${groupId}-${sectionId}-${createdAt}`,
-      createdAt,
-      content: {},
-      groupId,
-      sectionId,
+  describe("getLatestPublications", () => {
+    let student1;
+    let student2;
+    let student3;
+
+    let clazz: ClassModelType;
+
+    beforeEach(() => {
+      student1 = ClassStudentModel.create({
+        id: "1",
+        firstName: "aaa",
+        lastName: "aaa",
+        fullName: "aaa aaa",
+        initials: "AA"
+      });
+      student2 = ClassStudentModel.create({
+        id: "2",
+        firstName: "zzz",
+        lastName: "aaa",
+        fullName: "zzz aaa",
+        initials: "ZA"
+      });
+      student3 = ClassStudentModel.create({
+        id: "3",
+        firstName: "zzz",
+        lastName: "zzz",
+        fullName: "zzz zzz",
+        initials: "ZZ"
+      });
+
+      clazz = ClassModel.create({
+        name: "test",
+        classHash: "testHash",
+        students: [student1, student2, student3]
+      });
+    });
+
+    const getPublishedDocument = (createdAt: number, sectionId: string, uid: string) => {
+      return DocumentModel.create({
+        uid,
+        type: "publication",
+        key: `llDoc-${uid}-${sectionId}-${createdAt}`,
+        createdAt,
+        content: {},
+        groupId: "1",
+        sectionId,
+      });
+    };
+
+    it("finds documents from the correct section", () => {
```

Comment:
Previously we had one test checking several things - the correct section was returned, results were sorted etc. I prefer more modular tests, so especially now that sorting is more complex, I broke it into a few smaller tests.